rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs

```diff
@@ -288,3 +295,151 @@ fn test_start_and_stop_neuron_dissolve() {
+        let neuron_index = 0;
+        let from_subaccount = [0; 32];
+
+        env.rosetta_client
+            .create_neuron(
+                env.network_identifier.clone(),
+                &(*TEST_IDENTITY).clone(),
+                RosettaCreateNeuronArgs::builder(staked_amount.into())
+                    .with_from_subaccount(from_subaccount)
+                    .with_neuron_index(neuron_index)
+                    .build(),
+            )
+            .await
+            .unwrap();
+        // See if the neuron was created successfully
+        let agent = get_test_agent(env.pocket_ic.url().unwrap().port().unwrap()).await;
+
+        TransactionOperationResults::try_from(
+            env.rosetta_client
+                .start_dissolving_neuron(
+                    env.network_identifier.clone(),
+                    &(*TEST_IDENTITY).clone(),
+                    neuron_index,
+                )
+                .await
+                .unwrap()
+                .metadata,
+        )
+        .unwrap();
+
+        let mut neuron = list_neurons(&agent).await.full_neurons[0].to_owned();
+        // If we try to disburse the neuron when it is not yet DISSOLVE we expect an error
+        match env
+            .rosetta_client
+            .disburse_neuron(
+                env.network_identifier.clone(),
+                &(*TEST_IDENTITY).clone(),
+                RosettaDisburseNeuronArgs::builder(neuron_index)
+                    .with_recipient(TEST_IDENTITY.sender().unwrap().into())
+                    .build(),
+            )
+            .await
+        {
+            Err(e) if e.to_string().contains(&format!("Could not disburse: PreconditionFailed: Neuron {} has NOT been dissolved. It is in state Dissolving",neuron.id.unwrap().id)) => (),
+            Err(e) => panic!("Unexpected error: {}", e),
+            Ok(_) => panic!("Expected an error but got success"),
+        }
+        // Let rosetta catch up with the transfer that happended when creating the neuron
+        wait_for_rosetta_to_catch_up_with_icp_ledger(
+            &env.rosetta_client,
+            env.network_identifier.clone(),
+            &agent,
+        ).await;
+        let balance_before_disburse = env
+        .rosetta_client
+        .account_balance(
+            AccountBalanceRequest::builder(
+                env.network_identifier.clone(),
+                AccountIdentifier::from(TEST_IDENTITY.sender().unwrap()).into(),
+            )
+            .build(),
+        )
+        .await
+        .unwrap()
+        .balances
+        .first()
+        .unwrap()
+        .clone()
+        .value.parse::<u64>().unwrap();
+
+        // We now update the neuron so it is in state DISSOLVED
+        let now = env.pocket_ic.get_time().await.duration_since(UNIX_EPOCH).unwrap().as_secs();
+        neuron.dissolve_state = Some(DissolveState::WhenDissolvedTimestampSeconds(now - 1));
+        update_neuron(&agent, neuron.into()).await;
```

Comment:
I suppose this doesn't matter from the point of view of this test, but I believe in an earlier commit you were advancing the time and waiting for the governance canister to change the state of the neuron, whereas here you yourself in `update_neuron` make the call that the governance canister would otherwise make, i.e., change the state of the neuron to dissolved, is this correct? I'm curious if there was an issue doing it the other way, or maybe I misunderstood what you were trying to do?

Reply:
The advance time does not work because pocket ic and system time are not in sync. The ledger and governance canister use pocket ic time while rosetta uses system time.

Reply:
Ah, makes sense, thanks!